docs/src/main/sphinx/routines/examples.md

```diff
@@ -413,3 +413,112 @@ SELECT from_date_string('2023010123'); -- 2023-01-01 23:00:00.000 UTC (using the
+routine must end with a `RETURN` statement.
+
+```sql
+FUNCTION strtrunc(input VARCHAR)
+RETURNS VARCHAR
+BEGIN
+    CASE WHEN length(input) > 60
+    THEN
+        RETURN substr(input, 1, 30) || ' ... ' || substr(input, length(input) - 25);
+    ELSE
+        RETURN input;
+    END CASE;
+    RETURN input;
+END;
+```
+
+The next example changes over from a `CASE` to an `IF` statement, and avoids the
+duplicate `RETURN`:
+
+```sql
+FUNCTION strtrunc(input VARCHAR)
+RETURNS VARCHAR
+BEGIN
+    IF length(input) > 60 THEN
+        RETURN substr(input, 1, 30) || ' ... ' || substr(input, length(input) - 25);
+    END IF;
+    RETURN input;
+END;
+```
+
+All the preceding examples create the same output. Following is an example query
+which generates long strings to truncate:
+
+```sql
+WITH
+data AS (
+    SELECT substring('strtrunc truncates strings longer than 60 characters, leaving the prefix and suffix visible', 1, s.num) AS value
+    FROM table(sequence(start=>40, stop=>80, step=>5)) AS s(num)
+)
+SELECT
+    data.value
+  , strtrunc(data.value) AS truncated
+FROM data
+ORDER BY data.value;
+```
+
+The preceding query produces the following output with all variants of the
+routine:
+
+```
+                                      value                                       |                           truncated
+----------------------------------------------------------------------------------+---------------------------------------------------------------
+ strtrunc truncates strings longer than 6                                         | strtrunc truncates strings longer than 6
+ strtrunc truncates strings longer than 60 cha                                    | strtrunc truncates strings longer than 60 cha
+ strtrunc truncates strings longer than 60 characte                               | strtrunc truncates strings longer than 60 characte
+ strtrunc truncates strings longer than 60 characters, l                          | strtrunc truncates strings longer than 60 characters, l
+ strtrunc truncates strings longer than 60 characters, leavin                     | strtrunc truncates strings longer than 60 characters, leavin
+ strtrunc truncates strings longer than 60 characters, leaving the                | strtrunc truncates strings lon ... 60 characters, leaving the
+ strtrunc truncates strings longer than 60 characters, leaving the pref           | strtrunc truncates strings lon ... aracters, leaving the pref
+ strtrunc truncates strings longer than 60 characters, leaving the prefix an      | strtrunc truncates strings lon ... ers, leaving the prefix an
+ strtrunc truncates strings longer than 60 characters, leaving the prefix and suf | strtrunc truncates strings lon ... leaving the prefix and suf
+```
+
+A possible improvement is to introduce parameters for the total length:
+
+```sql
+FUNCTION strtrunc(input VARCHAR, count int)
+RETURNS VARCHAR
+BEGIN
+    DECLARE padding int;
+    SET padding = (count - 5) / 2;
```

Comment:
Rounding that happens here will cause the whole function to sometimes return strings of different length than count. I'd leave this exercise to the reader to make the width a parameter.

Reply:
I know .. still thought it would be helpful .. but fine .. I ok with removing as well

Reply:
Maybe we can explain the cascading of multiple routines in terms of signature in another example